Orange/widgets/data/oweditdomain.py

```diff
@@ -1555,6 +1565,18 @@ def get_data(self):
             tr.insert(0, self.format_cb.currentData())
         return var, tr
 
+    def _on_custom_change(self):
+        self.custom_text = self.custom_edit.text()
+        date_pat = r"%(-?)d|%(b|B)|%(-?)m|%(y|Y)|%(-?)j|%(-?)U|%(-?)W|%(a|A)|%w"
+        time_pat = r"%(-?)H|%(-?)I|%p|%(-?)M|%(-?)S|%f"
```

Comment:
If we allow arbitrary format, we could also allow arbitrary strings, like: "This happened in Year: %Y, Month: %m, Day: %d". I'd use `self.have_date = any(f"%{x}" in self.custom_text for x in "dBmyYUWwaA"`.